### PR TITLE
Fixed division by zero when setting "price" or "price override"

### PR DESCRIPTION
**Pull request by MahmoudRizk**
I want to merge this change because fix bug #4134

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

File: saleor/product/utils/costs.py [hunk 1]
```diff
@@ -58,5 +58,9 @@ def get_margin_for_variant(variant):
         return None
     base_price = variant.base_price
     margin = base_price - variant.cost_price
-    percent = round((margin / base_price) * 100, 0)
+    try:
+        percent = round((margin / base_price) * 100, 0)
+    except ZeroDivisionError:
+        sign = lambda margin: (1, -1)[margin.amount < 0]
+        percent = 9999999999999999999999 * sign(margin)
```
Comment (MahmoudRizk) [inline saleor/product/utils/costs.py]: since division any number by zero equals infinity, so 99999..99 here is to simulate infinity.

File: saleor/product/utils/costs.py [hunk 2]
```diff
@@ -58,5 +58,9 @@ def get_margin_for_variant(variant):
         return None
     base_price = variant.base_price
     margin = base_price - variant.cost_price
-    percent = round((margin / base_price) * 100, 0)
+    try:
```
Comment (maarcingebala) [inline saleor/product/utils/costs.py]: Hey @MahmoudRizk, thanks for opening the PR! I think it would make the most sense if we returned `None` in that case, similarly as we do if `cost_price` is also undefined. So the only change I would make is this:

```
base_price = variant.base_price
if not base_price:
    return None
```

**Comment by maarcingebala** [comment]
Looks good now! There is one last thing to take care of - could you add a test for that function, that would check that `None` is returned when product's or variant's price is zero? Unfortunately, I don't see any test for that function so you could add it in `tests/test_product.py`. Let me know if you can take care of that!

**Comment by MahmoudRizk** [comment]
> Looks good now! There is one last thing to take care of - could you add a test for that function, that would check that `None` is returned when product's or variant's price is zero? Unfortunately, I don't see any test for that function so you could add it in `tests/test_product.py`. Let me know if you can take care of that!

Ok, I will make it.

**Comment by maarcingebala** [comment]
Thanks for your contribution to Saleor @MahmoudRizk!
